src/miner.cpp

```diff
@@ -823,93 +826,118 @@ namespace pos {
 
-    const auto operatorName = args.operatorID.GetHex();
+    std::map<CKeyID, int32_t> nMinted;
+    std::map<CKeyID, int32_t> nTried;
 
     auto wallets = GetWallets();
 
-    while (true) {
-        boost::this_thread::interruption_point();
+    for (auto& arg : args) {
+        while (true) {
+            boost::this_thread::interruption_point();
 
-        bool found = false;
-        for (auto wallet : wallets) {
-            if (wallet->GetKey(args.operatorID, args.minterKey)) {
-                found = true;
+            bool found = false;
+            for (auto wallet : wallets) {
+                if (wallet->GetKey(arg.operatorID, arg.minterKey)) {
+                    found = true;
+                    break;
+                }
+            }
+            if (found) {
                 break;
             }
+            static std::atomic<uint64_t> time{0};
+            if (GetSystemTimeInSeconds() - time > 120) {
+                LogPrintf("ThreadStaker: unlock wallet to start minting...\n");
+                time = GetSystemTimeInSeconds();
+            }
+            std::this_thread::sleep_for(std::chrono::milliseconds(100));
         }
-        if (found) {
-            break;
-        }
-        static std::atomic<uint64_t> time{0};
-        if (GetSystemTimeInSeconds() - time > 120) {
-            LogPrintf("ThreadStaker (%s): unlock wallet to start minting...\n", operatorName);
-            time = GetSystemTimeInSeconds();
-        }
-        std::this_thread::sleep_for(std::chrono::seconds(1));
     }
 
-    LogPrintf("ThreadStaker (%s): started.\n", operatorName);
+    LogPrintf("ThreadStaker: started.\n");
 
-    while (true) {
+    while (!args.empty()) {
         boost::this_thread::interruption_point();
 
         while (fImporting || fReindex) {
             boost::this_thread::interruption_point();
 
-            LogPrintf("ThreadStaker (%s): waiting reindex...\n", operatorName);
+            LogPrintf("ThreadStaker: waiting reindex...\n");
 
             std::this_thread::sleep_for(std::chrono::milliseconds(900));
         }
 
-        try {
-            Staker::Status status = staker.stake(chainparams, args);
-            if (status == Staker::Status::error) {
-                LogPrintf("ThreadStaker (%s): terminated due to a staking error!\n", operatorName);
-                return nMinted;
-            }
-            if (status == Staker::Status::minted) {
-                LogPrintf("ThreadStaker (%s): minted a block!\n", operatorName);
-                nMinted++;
-            }
-            if (status == Staker::Status::initWaiting) {
-                LogPrintf("ThreadStaker (%s): waiting init...\n", operatorName);
+
+        int lastHeight = 0;
+        std::shared_ptr<CBlock> pblock;
+
+        for (auto it = args.begin(); it != args.end(); ) {
+            const auto& arg = *it;
+            const auto operatorName = arg.operatorID.GetHex();
+
+            pos::Staker staker;
```

Comment:
This definition can move outside of `for (auto it = args.begin(); it != args.end(); )`? Then can reuse the object.